### PR TITLE
create insight web component

### PR DESCRIPTION
**Pull request by bekand**
# Terms

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
>
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: cli/web/insight.html [hunk 1]
```diff
@@ -16,26 +16,15 @@
...
-
-    //Sending Function Name threw port
-    app.ports.receiveFunctionName.send("TestModel.Testing:add4");
-
-    //Sending Arguments List threw port
-    var argsList = [1,2,3,4];
-    app.ports.receiveFunctionArguments.send(argsList);
+    setTimeout(() => {
+        document.getElementById('i').init(distribution);
+    }, 5000);
```
Comment (AttilaMihaly) [inline cli/web/insight.html:27]: Why do we need this delay here?

File: cli/web/morphir-insight-element.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		this.app = Elm.Morphir.Web.Insight.init({
+    	node : this.shadowRoot.getElementById(MorphirInsightElement.divId),
+    	flags : {   distribution : this.distribution
+    	        ,   config : {  fontSize : 12 , decimalDigit : 2 }
+    	        }
+    	});
+
+		this.app.ports.receiveFunctionName.send(this.fqn);
+		this.app.ports.receiveFunctionArguments.send(this.arguments);
+	}
```
Comment (AttilaMihaly) [inline cli/web/morphir-insight-element.js:21]: The indentation seems to be off here. I find it hard to follow.

File: gulpfile.js [hunk 3]
```diff
@@ -85,6 +86,10 @@ function makeTryMorphir() {
     return make('cli', 'src/Morphir/Web/TryMorphir.elm', 'web/try-morphir.html')
 }
 
+async function addLitComponents() {
+    return src(['./cli/web/insight.js', './cli/web/morphir-insight-element.js']).pipe(concat('insight.js')).pipe(dest('./cli/web/'))
```
Comment (AttilaMihaly) [inline gulpfile.js:90]: What is concat doing here?

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

File: gulpfile.js (hunk 3 above)
Comment (bekand) [reply gulpfile.js:90]: The above line means: take this list of sources, concat them into a file named insight.js, and write it to this destination. It effectively takes insight.js and append the contents of morphir-insight-element.js

File: cli/web/insight.html (hunk 1 above)
Comment (bekand) [reply cli/web/insight.html:27]: It's just to test/show in the example insight.js app, that you can init the component after some arbitrary time has passed, no need to do it when the page is first loaded.

Comment (AttilaMihaly) [reply cli/web/insight.html:27]: Cool

**Review by AttilaMihaly — APPROVED** [review]
